impeller/entity/shaders/tiled_texture_fill.frag

```diff
@@ -21,11 +23,18 @@ in highp vec2 v_texture_coords;
 out f16vec4 frag_color;
 
 void main() {
-  frag_color =
-      IPHalfSampleWithTileMode(texture_sampler,   // sampler
-                               v_texture_coords,  // texture coordinates
-                               float16_t(frag_info.x_tile_mode),  // x tile mode
-                               float16_t(frag_info.y_tile_mode)   // y tile mode
-                               ) *
-      float16_t(frag_info.alpha);
+  if (supports_decal == 1.0) {
```

Comment:
Allows metal, vulkan, and most gles devices to use this shader at no additional cost over the texture_fill shader.